get_td_token.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# This script is a hacky helper that should be removed once we add support
+# for the TD oauth flow (acquiring initial token and refreshing)
+#
+# Usage:
+# TD_CONSUMER_KEY=<your_application_id> ./get_td_token.sh
+
+# DEPENDENCIES:
+#   - python3
+#   - curl
+#   - jq
+
+cd $(dirname $0)
+
+[[ -n "$TD_CONSUMER_KEY" ]] || { echo "Please set TD_CONSUMER_KEY"; exit 1; }
+
+TD_REDIRECT_URI=${TD_REDIRECT_URI:-https://127.0.0.1:42068/callback}
+
+uriencode() {
+    # cheating... i know...
+    python -c "import urllib.parse; print(urllib.parse.quote('''$1'''))"
+}
+
+uridecode() {
+    python -c "import urllib.parse; print(urllib.parse.unquote('''$1'''))"
+}
+
+get_new_auth() {
+    xdg-open "https://auth.tdameritrade.com/auth?response_type=code&redirect_uri=$(uriencode $TD_REDIRECT_URI)&client_id=$(uriencode $TD_CONSUMER_KEY)%40AMER.OAUTHAP"
```

Comment:
I know this is an example, but it'd be a good idea to limit the scope of the OAuth token:
```suggestion
    xdg-open "https://auth.tdameritrade.com/auth?response_type=code&redirect_uri=$(uriencode $TD_REDIRECT_URI)&client_id=$(uriencode $TD_CONSUMER_KEY)%40AMER.OAUTHAP&scope=AccountAccess"
```

Reply:
Oh this is fantastic, I was actually thinking I needed to look into this. Thanks!